cli/azd/cmd/auth_token.go

```diff
@@ -76,8 +86,36 @@ func (a *authTokenAction) Run(ctx context.Context) (*actions.ActionResult, error
 
 	var cred azcore.TokenCredential
 
+	// 1) flag --tenant-id is the highest priority. If it is not use, azd will check if subscriptionId is set as env var
+	tenantId := a.flags.tenantID
+	// 2) try to resolve tenant id when AZURE_SUBSCRIPTION_ID is set as system env
+	if tenantId == "" {
+		if subIdAtSystemEnv, found := os.LookupEnv(environment.SubscriptionIdEnvVarName); found {
+			if resolvedTenantId, err := a.subResolver.LookupTenant(ctx, subIdAtSystemEnv); err == nil {
```

Comment:
Let's make this experiecne better, by first checking for `guid` string. We might also want to support multiple guid formats.